### DISCUSSION TITLE
useQuery input is undefined

### DISCUSSION THREAD

**Discussion by NeurAlch** [trpc/trpc#2077] [Q&A] [answered]
Hey, I have this issue where all my `useQuery` inputs on React are marked as undefined. I'm using zod for the inputs/outputs

```
const doGetReport = trpc.useQuery(['report', { reportId: window.__id }, tasks], {
```

![image](https://user-images.githubusercontent.com/2329117/176988683-2552ec19-5eaf-4d92-90b5-bbc50ca9f41f.png)

Also my mutation variables are void:

![image](https://user-images.githubusercontent.com/2329117/176988748-3245f808-5a74-4994-a9f2-f584fa611f58.png)

I could not create a simple repo to reproduce it yet, but I'll try again next week, just wanted to know if anyone experienced this before and might have a known solution. I'm using 

```
    "@trpc/client": "^9.25.3",
    "@trpc/react": "^9.25.3",
    "@trpc/server": "^9.25.3",
    "zod": "^4.0.0-beta.1"
```

**Note**: had a similar issue with zod on latest non-beta

I even tried switching to yup, and had a similar issue, so got back to zod and @ts-ignore for now.

I don't think I'm doing anything out of the ordinary, but maybe? Here is some examples of the setup:

```
import { z } from 'zod';

export const inputDeleteKeywords = z.object({
  reportId: z.string(),
  keywords: z.array(z.string()),
});

export const outputDeleteKeywords = z.object({
  success: z.boolean(),
});
```

And

```
export const apiDeleteKeywords = async ({
  ctx,
  input,
}: {
  ctx: Context;
  input: z.input<typeof inputDeleteKeywords>;
}): Promise<z.output<typeof outputDeleteKeywords>> => { ... }
```

And

```
const doRemoveKeywords = trpc.useMutation('delete-keywords', {
    onSuccess: (data) => {
      if (data.success) {
        doGetItems.refetch();
      }
    },
  });
```

Any clue is appreciated, I'll try to create a reproducible example next week.

**Comment by kndwin** [comment, marked as answer]
I had a simliar issue and had to change the `strict: true` or `"strictNullChecks": true` option in the `package.json` 
and it started to work (my other types broke though but that's okay)

**Comment by NeurAlch** [comment]
Just double checked and just realized my `tsconfig.json` file had `strict: true` but in another part of the file it had `strictNullChecks: false` for some reason...
